### ISSUE TITLE
nodes in testnet created via launcher are out of sync

### ISSUE THREAD

**Issue by pavybez** [EOSIO/eos#339]
We created and started two local nodes in testnet using launcher and created an account on one of them. But we can only query the account information on one of them even after the blocks catch up on both of them. Screen shot below:

![image](https://user-images.githubusercontent.com/7913634/30051530-f438bcb4-9254-11e7-95ab-275879efd4f8.png)

**Comment by pavybez** [comment]
Additionally we checked that they have different blocks:

![image](https://user-images.githubusercontent.com/7913634/30053193-83db5886-925a-11e7-9101-beb38e90a5f5.png)

**Comment by pmesnier** [comment]
Thank you for your report. This is an area that is rapidly improving. When whas the last time you updated your code? Just yesterday some relevant changes were merged into the master branch, 

I just tested both scenarios here on a 4-node testnet I'm working with and the transactions were duplicated   just fine. 

I will be pushing more changes later today as well.

**Comment by sergmetelin** [comment]
It was built off today's fresh build, not syncing for us

**Comment by pmesnier** [comment]
ok. I should have some updates ready later today. Attached are a couple of test run results showing the expected synchronization:
[dupe2.txt](https://github.com/EOSIO/eos/files/1277063/dupe2.txt)
[dupe1.txt](https://github.com/EOSIO/eos/files/1277064/dupe1.txt)

**Comment by pmesnier** [comment]
Also can you show me the exact steps you used to launch and populate your failing test?

**Comment by pavybez** [comment]
Hi,

We are running the test on ubuntu. These are the steps we followed:

1. Build eos
2. Launch testnet using launcher as ./programs/launcher/launcher -n2 -p2 -l local
3. Launch eos-walled on port 7777
4. Use eosc to create a wallet as ./eosc --wallet-port 7777 wallet create
5. Import private key of inita as ./eosc --wallet-port 7777 wallet import <key>
6. Use eosc to create a currency account as ./eosc --wallet-port 7777 create account inita currency <key> <key>
7. Use eosc to query for the newly created account on a different eosd (not 8888) as 
    ./esoc -p 8889 get account currency.

This fails.

Also in the stderr.txt I noticed the following lines:

1433705ms thread-0   net_plugin.cpp:337            connect              ] host: localhost port: 9877
1433705ms thread-0   net_plugin.cpp:1032           operator()           ] caught signal 0
1433708ms thread-0   net_plugin.cpp:138            connection           ] created connection from client


Apparently it catches a signal and exits the asio.

**Comment by pavybez** [comment]
On a similar note, I noticed that in net_plugin.cpp, in the method net_plugin::plugin_startup(..) method the following lines:

boost::asio::signal_set signals (app().get_io_service(), SIGINT, SIGTERM);
    signals.async_wait ([this](const boost::system::error_code &ec, int signum) {
        dlog ("caught signal ${sn}", ("sn", signum) ) ;
      });


My question is since signals is a local variable on the stack, wouldn't it be destroyed when this startup function returns?

Best regards,
Pravin

**Comment by jgiszczak** [comment]
Corrected problems in the network code.  Waiting on Phil before merging and committing.

**Comment by pmesnier** [comment]
To Pravin's question, the signal handler is getting removed, there's already one in the app library that works. Also, Jonathan is going to put his fix in its own branch so it can be committed separate from mine.

**Comment by heifner** [comment]
This can be closed.  What it specifically addressed has been taken care of, and more general issues have their own tickets.
